crates/miden-objects/src/transaction/inputs.rs

```diff
@@ -531,3 +535,170 @@ pub fn validate_account_seed(
         (false, None) => Ok(()),
     }
 }
+
+// FOREIGN ACCOUNT INPUTS
+// ================================================================================================
+
+/// Contains information about a foreign account, with everything required to execute its code.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ForeignAccountInputs {
+    /// Account header of the foreign account.
+    account_header: AccountHeader,
+    /// Header information about the account's storage.
+    storage_header: AccountStorageHeader,
+    /// Code associated with the account.
+    account_code: AccountCode,
+    /// Merkle proof of the account's inclusion in the account tree.
+    account_witness: MerklePath,
```

Comment:
I think they type would now need to be `AccountWitness`. I'd also probably name it just `witness`.